src/coreclr/jit/gentree.cpp

```diff
@@ -21380,12 +21422,15 @@ GenTree* Compiler::gtNewSimdCvtNode(var_types   type,
 
 #if defined(TARGET_XARCH)
     assert(IsBaselineVector512IsaSupportedDebugOnly() ||
+           (simdSize != 64 && compIsaSupportedDebugOnly(InstructionSet_AVX10v1)) ||
            ((simdTargetBaseType == TYP_INT) && ((simdSize == 16 && compIsaSupportedDebugOnly(InstructionSet_SSE41)) ||
                                                 (simdSize == 32 && compIsaSupportedDebugOnly(InstructionSet_AVX)))));
 
     GenTree* fixupVal;
+    bool isAvx10v1 = false;
 
-    if (IsBaselineVector512IsaSupportedOpportunistically())
+    if (IsBaselineVector512IsaSupportedOpportunistically() ||
```

Comment:
I don't think we need the `simdSize != 64` check. We should only be encountering `simdSize == 64` if `Avx512` is supported; otherwise the JIT should never see TYP_SIMD64 on xarch.

I think we can take advantage of that to simplify the real check and instead have a simple assert in the `Avx10` path that handles things instead.

Reply:
Similar to this, https://github.com/dotnet/runtime/pull/101938#discussion_r1607357803, are you asking to get rid of the ```if (IsBaselineVector512IsaSupportedOpportunistically() || (simdSize != 64 && (isAvx10v1 = compOpportunisticallyDependsOn(InstructionSet_AVX10v1))))``` and do something like ```if(canUseEvexEncoding())``` and then do a separate check for ```compOpportunisticallySupports``` for Avx10 and handle it? 

Shall I not try to reduce the duplicate ```compOpportunisticallySupports(InstructionSet_AVX10v1)``` and rely on compiler optimization or use ```canUseEvenExcoding()``` and also check for ```Avx10v1``` support inside?